compiler/rustc_lint/src/unnecessary_send_constraint.rs

```diff
@@ -0,0 +1,56 @@
+use rustc_span::sym;
+
+use crate::hir;
+
+use crate::{lints::UnnecessarySendConstraintDiag, LateContext, LateLintPass};
+
+declare_lint! {
+    /// The `lint_unnecessary_send_constraint` lints unnecessary constraint of references to `Send`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// fn foo(_: &(dyn Any + Send>) {}
```

Comment:
This asserts compile_fail for a typo, while the lint itself is only WARN. Do you mean
```suggestion
    /// #[deny(unnecessary_send_constraint)]
    /// fn foo(_: &(dyn Any + Send)) {}
```
?